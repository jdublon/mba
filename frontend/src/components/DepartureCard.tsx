import { DepartureCardProps } from "@/types";
import { FC } from "react";

export const DepartureCard: FC<DepartureCardProps> = ({ departure, label }) => {
  const textStyles = "font-sans text-primary-green text-sm font-semibold";
  const availabilityText =
    departure.available_pax > 0
      ? "Spaces available"
      : "Sorry this trip is fully booked!";

  return (
    <div key={label} className="p-4 border rounded-lg shadow-md bg-white">
      <div className="space-y-2">
        <div className={textStyles}>Start Date: {departure.start_date}</div>
        <div className={textStyles}>Price: Â£{`${departure.price}`}</div>
        <div className={textStyles}>Availability: {availabilityText}</div>
      </div>
    </div>
  );
};
