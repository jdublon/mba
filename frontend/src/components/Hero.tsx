import { FC } from "react";
import { IconLabel } from "./IconLabel";

export const Hero: FC = () => {
  return (
    <div className="relative w-full h-screen overflow-hidden">
      <div
        className="absolute top-0 left-0 right-0 bottom-0 
          bg-gradient-to-t 
          from-black via-black to-transparent 
          bg-[linear-gradient(0deg,_#2A2D2C_0.09%,_rgba(42,_45,_44,_0)_59.97%)] 
          bg-cover bg-center bg-no-repeat 
          bg-[url('/images/hero-image-sm.jpg')] 
          sm:bg-[url('/images/hero-image-sm.jpg')] 
          md:bg-[url('/images/hero-image-md.jpg')] 
          lg:bg-[url('/images/hero-image-lg.jpg')]"
      />

      <div className="absolute bottom-0 left-0 right-0 inline-flex flex-col justify-start items-start">
        <div className="mx-4 mb-3 sm:mb-[1.62rem]">
          <h1 className="text-white text-[2rem] sm:text-[3rem] font-extrabold font-sans leading-9 sm:leading-[3.375rem] tracking-[-0.04rem] mb-6 sm:mb-[1.62rem]">
            ISLAND HOPPING ADVENTURE IN THE GALAPAGOS
          </h1>
          <div className="flex gap-6 flex-wrap">
            <IconLabel icon="location" label="Galapagos islands, Ecuador" />
            <IconLabel icon="duration" label="9 nights" />
            <IconLabel icon="difficulty" label="Easy" />
          </div>
        </div>

        <div className="bg-black sm:bg-transparent w-screen px-4 pt-3 sm:pt-0">
          <p className="text-white font-serif text-base leading-[1.375rem] tracking-[-0.02rem] pb-10 sm:pb-12">
            Hike, bike and snorkel this unique archipelago and encounter its
            special wildlife, on an adventure immersed in the natural world
          </p>
        </div>
      </div>
    </div>
  );
};
