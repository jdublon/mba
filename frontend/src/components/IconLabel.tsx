import { FC } from "react";
import Image from "next/image";
import { IconLabelProps } from "@/types";

export const IconLabel: FC<IconLabelProps> = ({ icon, label }) => (
  <div className="flex items-center">
    <Image
      src={`/icons/${icon}.svg`}
      alt={icon}
      width={16}
      height={16}
      className="mr-[0.5rem]"
    />
    <span className="text-white text-[0.75rem] sm:text-[0.875rem] font-extrabold font-sans leading-[1rem] sm:leading-[1.125rem] tracking-[-0.015rem] sm:tracking-[-0.0175rem] font-sans uppercase">
      {label}
    </span>
  </div>
);
