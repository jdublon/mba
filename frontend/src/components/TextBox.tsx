import { FC } from "react";
import { TextBoxProps } from "@/types";

export const TextBox: FC<TextBoxProps> = ({
  text,
  textStyles,
  containerStyles,
}) => (
  <div className={containerStyles}>
    <p className={textStyles}>{text}</p>
  </div>
);
