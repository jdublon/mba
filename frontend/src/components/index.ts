export { DepartureCard } from "./DepartureCard";
export { Hero } from "./Hero";
export { IconLabel } from "./IconLabel";
