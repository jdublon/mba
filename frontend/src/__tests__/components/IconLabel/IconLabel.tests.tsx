import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import { IconLabel } from "@/components/IconLabel";
import { mockIcons } from "./fixtures";

describe("Component: IconLabel", () => {
  it("SHOULD match snapshot WHEN rendered", () => {
    const component = render(<IconLabel icon="location" label="Space" />);
    expect(component).toMatchSnapshot();
  });

  it.each(mockIcons)(`SHOULD render UI as expected for each icon`, (icon) => {
    const component = render(<IconLabel icon={icon.icon} label={icon.label} />);
    expect(component.getByAltText(icon.icon));
    expect(component.getByText(icon.label));
  });
});
