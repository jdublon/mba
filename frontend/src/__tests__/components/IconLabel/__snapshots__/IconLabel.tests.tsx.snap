// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: IconLabel SHOULD match snapshot WHEN rendered 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex items-center"
      >
        <img
          alt="location"
          class="mr-[0.5rem]"
          data-nimg="1"
          decoding="async"
          height="16"
          loading="lazy"
          src="/icons/location.svg"
          style="color: transparent;"
          width="16"
        />
        <span
          class="text-white text-[0.75rem] sm:text-[0.875rem] font-extrabold font-sans leading-[1rem] sm:leading-[1.125rem] tracking-[-0.015rem] sm:tracking-[-0.0175rem] font-sans uppercase"
        >
          Space
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex items-center"
    >
      <img
        alt="location"
        class="mr-[0.5rem]"
        data-nimg="1"
        decoding="async"
        height="16"
        loading="lazy"
        src="/icons/location.svg"
        style="color: transparent;"
        width="16"
      />
      <span
        class="text-white text-[0.75rem] sm:text-[0.875rem] font-extrabold font-sans leading-[1rem] sm:leading-[1.125rem] tracking-[-0.015rem] sm:tracking-[-0.0175rem] font-sans uppercase"
      >
        Space
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
